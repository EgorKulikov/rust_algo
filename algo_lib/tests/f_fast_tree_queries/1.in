5 6
1 2
1 3
3 4
3 5
? 2 5
+ 1 4 1
? 2 5
+ 4 5 2
? 4 5
? 1 1
