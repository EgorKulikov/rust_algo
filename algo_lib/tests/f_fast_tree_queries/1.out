5
1
6
2
